src/rulesets/Base/base.jl

```diff
@@ -168,11 +168,18 @@ end
 # note: rules for ^ are defined in the fastmath_able.jl
 function frule((_, _, Δx, _), ::typeof(Base.literal_pow), ::typeof(^), x::Real, pv::Val{p}) where p
     y = Base.literal_pow(^, x, pv)
-    return y, (p * y / x * Δx)
+    return y, ifelse(iszero(x), zero(y), p * y / x * Δx)
```

Comment:
This could cause type-instability if `Δx` is not the same type as `x`. Since the point of `y/x` is to avoid computing `x^(p-1)`, perhaps something like this? The additional `oftype` also ensures that if `x`, `Δx`, and `p` are all integers, we don't unnecessarily make everything a float.
```suggestion
    xpm1 = y / x
    return y, p * oftype(y, ifelse(iszero(x), zero(xpm1), xpm1)) * Δx
```
What do you think?

Reply:
Another alternative:
```suggestion
xpm1 = if y isa Integer
    iszero(x) ? zero(y) : div(y, x)
else
    iszero(x) ? zero(y) : y / x
end
return y, p * xpm1 * Δx
```
I'm not certain how the use of the ternary vs `ifelse` would impact performance.